physicsnemo/models/module.py

```
@@ -461,8 +483,7 @@ def from_checkpoint(
                 local_path.joinpath("model.pt"), map_location=model.device
             )
 
-            model_dict = convert_ckp_apex(ckp_args, model_args, model_dict)
-            model.load_state_dict(model_dict, strict=False)
+            load_state_dict_with_logging(model, model_dict, strict=False)
```

I'm generally uncomfortable with `strict=False` here. I realize it was injected in #809 rather than this PR, but since this is a widely used function affecting all trained models in physicsnemo, can we revert it to `strict=True` here? Seems like backwards-compat handling should be taken care of by the time this line is run. Thoughts @CharlelieLrt ?

---

@pzharrington good point! I was actually wondering why we were setting `strict=False` here, but I thought it was there since the beginning and I've never realized it was introduced by #809 . We should definitely revert it back to `True`. AFAIK it's only useful when fine-tuning some parts of the model, or other things like this...

@juliusberner does your application requires `strict=False` there, or would it be okay to revert to `True`?

---

Probably also worth checking with @LostnEkko -- was it introduced in #809 as part of the apex checkpoint handling, and would you be able to test your use-case off of this PR to see if `strict=True` works?

---

I think the best option would be to expose it (similar as in the `load` method above). We can then default it to `True` if we want --- if the user sets it to `False`, we would still log unexpected/missing keys with this PR (preventing silent errors which are happening right now since the checkpoint conversion has a bug in https://github.com/NVIDIA/physicsnemo/blob/d1c9391f0f594f7279c8990bff70b8227a6d1f93/physicsnemo/models/util_compatibility.py#L92C17-L92C50)